python/morpheus/morpheus/_lib/src/stages/filter_detections.cpp

```diff
@@ -20,30 +20,32 @@
 #include "mrc/segment/builder.hpp"  // for Builder
 #include "mrc/segment/object.hpp"   // for Object
 
-#include "morpheus/messages/control.hpp"           // for ControlMessage
-#include "morpheus/messages/multi.hpp"             // for MultiMessage
-#include "morpheus/messages/multi_tensor.hpp"      // for MultiTensorMessage
-#include "morpheus/objects/dev_mem_info.hpp"       // for DevMemInfo
-#include "morpheus/objects/dtype.hpp"              // for DType
-#include "morpheus/objects/memory_descriptor.hpp"  // for MemoryDescriptor
-#include "morpheus/objects/table_info.hpp"         // for TableInfo
-#include "morpheus/types.hpp"                      // for RangeType
-#include "morpheus/utilities/matx_util.hpp"        // for MatxUtil
-#include "morpheus/utilities/tensor_util.hpp"      // for TensorUtils
+#include "morpheus/messages/control.hpp"               // for ControlMessage
+#include "morpheus/messages/memory/tensor_memory.hpp"  // for TensorMemory
+#include "morpheus/messages/meta.hpp"                  // for MessageMeta
+#include "morpheus/objects/dev_mem_info.hpp"           // for DevMemInfo
+#include "morpheus/objects/dtype.hpp"                  // for DType
+#include "morpheus/objects/memory_descriptor.hpp"      // for MemoryDescriptor
+#include "morpheus/objects/table_info.hpp"             // for TableInfo
+#include "morpheus/objects/tensor_object.hpp"          // for TensorObject
+#include "morpheus/types.hpp"                          // for RangeType
+#include "morpheus/utilities/matx_util.hpp"            // for MatxUtil
+#include "morpheus/utilities/tensor_util.hpp"          // for TensorUtils
 
 #include <cuda_runtime.h>                         // for cudaMemcpy, cudaMemcpyKind
 #include <cudf/column/column_view.hpp>            // for column_view
 #include <cudf/types.hpp>                         // for data_type
 #include <glog/logging.h>                         // for COMPACT_GOOGLE_LOG_FATAL, LogMessageFatal, CHECK, DCHECK
 #include <mrc/cuda/common.hpp>                    // for MRC_CHECK_CUDA
 #include <rmm/cuda_stream_view.hpp>               // for cuda_stream_per_thread
+#include <rmm/device_buffer.hpp>                  // for device_buffer
 #include <rmm/mr/device/per_device_resource.hpp>  // for get_current_device_resource
 
 #include <cstddef>     // for size_t
 #include <cstdint>     // for uint8_t
 #include <exception>   // for exception_ptr
 #include <functional>  // for function
-#include <memory>      // for make_shared, shared_ptr, __shared_ptr_access
+#include <memory>      // for shared_ptr, __shared_ptr_access, make_shared
```

Comment:
Remove `__shared_ptr_access` we shouldn't be calling out implementation detail symbols.

```suggestion
#include <memory>      // for shared_ptr, make_shared
```